src/users/users.service.ts

```diff
@@ -43,9 +43,62 @@ export class UsersService {
   }
 
   async createUser(email: string, walletAddress: string): Promise<User> {
+    const emailExists = await this.emailAlreadyUsed(email);
+    if (emailExists) {
+      throw new UsersFieldError(
+        UsersFieldError.EMAIL_CODES.ALREADY_EXIST,
+        'The email address is not unique and already exist',
+        { email: 'Alread used' },
+      );
+    }
     const user = await this.prisma.user.create({
       data: { email, walletAddress },
     });
     return User.create(user);
   }
```

Comment:
This is redundant with the `updateUser` method. Could be an `updateOrCreateUser` instead. Also, it's brittle, if we change the conditions about creating a user or updating one, we might forget to update one or the other.

Reply:
i did it because the db schema changed and we dont have any way to validate other than with a function that the email already is used. as the email is now nullable in our model, we cannot set it to @unique. creating an updateOrCreate method would add more complexity as we would have to deal with a flexible function that is not fited for the moment IMHO. i would prefer to keep it separated (updateUser is in fact updateEmail as we dont update any other field rn).